### PR TITLE
Update readme to include version and compatiblity information

### PR DESCRIPTION
**Pull request by BenWhitehead**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -58,3 +58,107 @@ export ZOOKEEPER_SECRET <secret>
...
+
+|       | application/zip (version-2.x) | application/vnd.dcos.universe.repo+json;charset=utf-8;version=v3 |
+| ----- | ----------------------------- | ---------------------------------------------------------------- |
+| 0.1.0 | Not Supported                 | Not Supported                                                    |
+| 0.1.1 | Not Supported                 | Not Supported                                                    |
+| 0.1.2 | Not Supported                 | Not Supported                                                    |
+| 0.1.3 | Not Supported                 | Not Supported                                                    |
+| 0.1.4 | Not Supported                 | Not Supported                                                    |
+| 0.1.5 | Not Supported                 | Not Supported                                                    |
+| 0.2.0 | Not Supported                 | Supported                                                        |
```
Comment (cruhland) [inline README.md]: Shouldn't the `application/zip` column have "Supported" for all entries?

Comment (cruhland) [inline README.md]: Also, to make the table less of a chore to read through, could we summarize it into two rows, `0.1.x` and `0.2.0`?

File: README.md [hunk 2]
```diff
@@ -58,3 +58,107 @@ export ZOOKEEPER_SECRET <secret>
...
+
+|       |    2.0    |      3.0      |
+| ----- | --------- | ------------- |
+| 0.1.0 | Supported | Not Supported |
+| 0.1.1 | Supported | Not Supported |
+| 0.1.2 | Supported | Not Supported |
+| 0.1.3 | Supported | Not Supported |
+| 0.1.4 | Supported | Not Supported |
+| 0.1.5 | Supported | Not Supported |
+| 0.2.0 | Supported | Supported     |
```
Comment (cruhland) [inline README.md]: Can this table be summarized to `0.1.x` and `0.2.0` rows as well?

File: README.md [hunk 3]
```diff
@@ -58,3 +58,107 @@ export ZOOKEEPER_SECRET <secret>
...
+
+### API Method Version Compatibility
+
+The following requests have constraints based on the version of the package from universe
```
Comment (cruhland) [inline README.md]: From the word "constraints" it sounds like we're really trying to convey when the endpoints will *not* work. If that's the case, this section could be made much shorter by documenting only the endpoint limitations, and stating that everything else will work.

For example, for `/package/install` we only need to say the following:

```
Content-Type: application/vnd.dcos.package.install-request+json;charset=utf-8;version=v1
Accept:       application/vnd.dcos.package.install-response+json;charset=utf-8;version=v1
```
A v1 install can fail when the package being installed was published as a Universe package with `packagingVersion` 3.0 and the package has no marathon template defined.

File: README.md [hunk 4]
```diff
@@ -58,3 +58,107 @@ export ZOOKEEPER_SECRET <secret>
...
+
+#### `/package/describe`
+
+```
+Conent-Type: application/vnd.dcos.package.describe-request+json;charset=utf-8;version=v1
```
Comment (cruhland) [inline README.md]: Typo in `Conent-Type` -- applies to all occurrences below.

File: README.md [hunk 5]
```diff
@@ -58,3 +58,107 @@ export ZOOKEEPER_SECRET <secret>
...
+Accept:      application/vnd.dcos.package.install-response+json;charset=utf-8;version=v1
+```
+A v1 install can succeed in the following scenarios:
+
+1. The package being installed was published as a Universe package with `packagingVersion` 2.0
+2. The package being installed was published as a Universe package with `packagingVersion` 3.0 and the package has a marathon template defined
+
+```
+Conent-Type: application/vnd.dcos.package.install-request+json;charset=utf-8;version=v1
+Accept:      application/vnd.dcos.package.install-response+json;charset=utf-8;version=v1
```
Comment (cruhland) [inline README.md]: This should have `version=v2`.

File: README.md [hunk 6]
```diff
@@ -58,3 +58,107 @@ export ZOOKEEPER_SECRET <secret>
...
+```
+A v2 install can succeed in the following scenarios:
+
+1. The package being installed was published as a Universe package with `packagingVersion` 2.0
+2. The package being installed was published as a Universe package with `packagingVersion` 3.0 and the package has a marathon template defined
+3. The package being installed was published as a Universe package with `packagingVersion` 3.0 and the package has a `.cli` object defined in it's resource set
```
Comment (cruhland) [inline README.md]: Here's a good example where saying what can fail would be easier to interpret. In this case, the only scenario that can fail is a `packagingVersion` 3.0 package that has neither a marathon template nor a `.cli` object, correct? Or will even that case succeed?

File: README.md (hunk 1 above)
Comment (BenWhitehead) [inline README.md]: Good catch, editing error. Will update.

File: README.md (hunk 2 above)
Comment (BenWhitehead) [inline README.md]: sure

**Comment by cruhland** [comment]
@BenWhitehead back to you.

**Comment by cruhland** [comment]
@BenWhitehead LGTM

**Pull request merged**